front/app/modules/free/user_confirmation/citizen/components/ConfirmationSignupStep.tsx

```diff
@@ -0,0 +1,375 @@
+  width: 100%;
+  max-width: ${(props) => (props.inModal ? '380px' : 'unset')};
+  display: flex;
+  flex-direction: column;
+  align-items: stretch;
+  margin: 2rem 0;
+`;
+
+export const FormField = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: stretch;
+`;
+
+export const StyledLabel = styled(Label)`
+  display: block;
+  text-align: center;
+`;
+
+export const LabelTextContainer = styled.div`
+  display: block;
+  color: ${({ theme }) => theme.colorText};
+  margin-bottom: 1.5rem;
+  font-size: ${fontSizes.base};
+
+  &:last-child {
+    margin-bottom: 1rem;
+  }
+`;
+
+export const Footer = styled.div`
+  display: flex;
+  justify-content: center;
+  padding-top: 10px;
+`;
+
+export const CancelButton = styled(Button)``;
+
+export const HelpImage = styled.img`
+  width: 100%;
+`;
+
+export const SubmitButton = styled(Button)`
+  width: 100%;
+  margin: 0.75rem 0;
+`;
+
+const FooterNotes = styled.div`
+  margin: 18px 0 0;
+  text-align: center;
+`;
+
+const FooterNote = styled.p`
+  color: ${({ theme }) => theme.colorText};
+  font-size: ${fontSizes.small}px;
+  line-height: normal;
+
+  &:not(:last-child) {
+    margin: 0 0 1rem;
+  }
+`;
+
+const FooterNoteLink = styled(Link)`
+  font-size: ${fontSizes.small}px;
+  padding-left: 4px;
+  color: ${({ theme }) => theme.colorText};
+  text-decoration: underline;
+
+  &:hover {
+    color: ${({ theme }) => darken(0.2, theme.colorText)};
+    text-decoration: underline;
+  }
+`;
+
+const FooterNoteSuccessMessage = styled.span`
+  color: ${colors.clGreen};
+  padding-left: 6px;
+`;
+
+const FooterNoteSuccessMessageIcon = styled(Icon)`
+  width: 12px;
+  height: 12px;
+  margin-right: 4px;
+`;
+
+type Props = SignUpStepOutletProps & InjectedIntlProps;
+
+const ConfirmationSignupStep = ({
+  metaData,
+  intl: { formatMessage },
+  onCompleted,
+  ...props
+}: Props) => {
+  const user = useAuthUser();
+  const [confirmation, setConfirmation] = useState<IConfirmation>({
+    code: null,
+  });
+  const [newEmail, setNewEmail] = useState<string | null | undefined>(
+    undefined
+  );
+  const [apiErrors, setApiErrors] = useState<CLErrors>({});
+  const [processing, setProcessing] = useState<boolean>(false);
+  const [changingEmail, setChangingEmail] = useState<boolean>(false);
+  const [codeResent, setCodeResent] = useState<boolean>(false);
+
+  useEffect(() => {
+    props.onData({
+      key: 'confirmation',
+      configuration: {
+        position: 2.1,
+        stepName: formatMessage(messages.confirmYourAccount),
+        onSkipped: () => trackEventByName(tracks.signUpConfirmationStepSkipped),
+        onError: () => trackEventByName(tracks.signUpConfirmationStepFailed),
+        onCompleted: () =>
+          trackEventByName(tracks.signUpConfirmationStepCompleted),
+        isEnabled: (metaData) => !!metaData?.requiresConfirmation,
+        isActive: (authUser) => !authUser?.attributes?.email_confirmed_at,
+      },
+    });
+  }, []);
```

Comment:
Would add `onData`, `metaData` and `authUser` to the `useEffect` dependency array to be on the safe side.

Reply:
This for some reason creased an infitine rendering loop, not sure which property is causing it, but in a similar component the dependency array is also empty so I might leave it as is